### PR TITLE
Use gha-puppet again

### PR DESCRIPTION
**Pull request by ekohl**
This is to test out some new workflows where we again rely on the common workflows but while also adding the option to test multiple versions.

Right now it's unknown if this concept works.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -1,100 +1,20 @@
...
-      - run: echo Test suite completed
+jobs:
+  puppet:
+    name: Puppet
+    uses: ekohl/gha-puppet/.github/workflows/beaker.yml@set-in-env-var
+    with:
+      pidfile_workaround: 'false'
+      rubocop: false
+      cache-version: '1'
+      beaker_facter: '--beaker-facter pulpcore_version:Pulp:3.28,3.22,3.21'
```
Comment (evgeni) [inline .github/workflows/ci.yml:20]: this feels odd api wise.
either: `m2gha_options: --beaker-facter pulpcore_version:Pulp:3.28,3.22,3.21`
or: `beaker_facter: ' pulpcore_version:Pulp:3.28,3.22,3.21'`
wdyt?

**Comment by ekohl** [comment]
For complete context, there are now PRs up: https://github.com/voxpupuli/puppet_metadata/pull/108 & https://github.com/voxpupuli/gha-puppet/pull/42

**Comment by ekohl** [comment]
Note to self: also support `beaker_facter` in `.sync.yml` so the file can be managed again.

**Comment by ekohl** [comment]
Something is different with curl: `curl: unknown --write-out variable: 'stderr`.

When I run it manually on EL8:

```console
# curl --silent --write-out '%{stderr}Response-Code: %{response_code}\n' https://ekohl.nl
<html>
<head><title>301 Moved Permanently</title></head>
<body>
<center><h1>301 Moved Permanently</h1></center>
<hr><center>nginx</center>
</body>
</html>
curl: unknown --write-out variable: 'stderr'
Response-Code: 301
# curl --silent --write-out '%{stderr}Response-Code: %{response_code}\n' https://ekohl.nl > /dev/null 
curl: unknown --write-out variable: 'stderr'
# curl --silent --write-out '%{stderr}Response-Code: %{response_code}\n' https://ekohl.nl 2> /dev/null 
<html>
<head><title>301 Moved Permanently</title></head>
<body>
<center><h1>301 Moved Permanently</h1></center>
<hr><center>nginx</center>
</body>
</html>
Response-Code: 301
```

So now I wonder how this ever worked. https://github.com/theforeman/puppet-pulpcore/pull/311 is passing, but now I can't explain why.

**Comment by evgeni** [comment]
Great detective work!

**Pull request closed without merging**